### PR TITLE
chore: Add rules for apply section && test: Add tests for apply section

### PR DESCRIPTION
**Pull request by HashirGJ8842**

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -223,11 +235,14 @@ service cloud.firestore {
...
     match /members/{membersId} {
-      allow read, create, update: if isAdmin();
+      allow read, delete: if isAdmin();
+      allow create: if documentFieldsCheckOut(['timestamp', 'course', 'email', 'paymentStatus', 'enrollmentNumber', 'facultyNumber', 'mobile', 'name', 'registrationNumber'], []) && memberFieldsTypeValidate(request.resource.data);
+      allow update: if isAdmin() && editOnlyChangesField(['timestamp', 'course', 'email', 'paymentStatus', 'enrollmentNumber', 'facultyNumber', 'mobile', 'name', 'registrationNumber']) && memberFieldsTypeValidate(request.resource.data);
     }
 
     match /facultyNumbers/{facultyNumbersId} {
-      allow read, create, update: if isAdmin();
+      allow read, write;
```
Comment (iamareebjamal) [inline firestore.rules:244]: Why allowing write to anyone?

**Review by Rishabhltfb — APPROVED** [review]
Ok from my side, Once Areeb Bhai approves your changes in facultyNumber then can merge.

**Comment by Rishabhltfb** [comment]
Please add news and facultyNumber collection tests as well before we merge.

**Review by Rishabhltfb — CHANGES_REQUESTED** [review]
Unit Tests Fail, Please fix them.

**Comment by HashirGJ8842** [comment]
@Rishabhltfb, Fixed it

File: firestore.rules (hunk 1 above)
Comment (Rishabhltfb) [inline firestore.rules]: `write`  consists of all three `create`, `update` and `delete`, so if you once allow write to everyone then it'll not check next line i.e `allow delete, update: if isAdmin();` 
Please fix.

**Review by Rishabhltfb — CHANGES_REQUESTED** [review]

File: firestore.rules (hunk 1 above)
Comment (HashirGJ8842) [reply firestore.rules]: @Rishabhltfb, this change is outdated. Please review the latest changes :sweat_smile:

**Review by Rishabhltfb — APPROVED** [review]
LGTM

**Pull request merged**